packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -202,15 +204,45 @@ export default class InternalModel {
     // `lastObject` have changed.  When this happens we don't want those
     // models to rematerialize their records.
 
+    // eager checks to avoid instantiating record data if we are empty or loading
+    if (this.isEmpty()) {
+      return true;
+    }
+
+    if (this.isLoading()) {
+      return false;
+    }
+
+    let isRecordFullyDeleted;
+    if (ENABLE_RECORD_DATA_STATE) {
+      isRecordFullyDeleted = this._isRecordFullyDeleted();
+    } else {
+      isRecordFullyDeleted = this.currentState.stateName === 'root.deleted.saved';
+    }
     return (
-      this._isDematerializing ||
       this.hasScheduledDestroy() ||
+      this._isDematerializing ||
```

Comment:
why the ordering switch?

Reply:
debugging leftover, reverting to original